src/controllers/user.js

```diff
@@ -12,7 +14,56 @@ const getProfile = async (req, res) => {
         .json(currentUser);
     }
   } catch (error) {
-    return res.sendStatus(500).json({ message: error.message });
+    return res.status(500).json({ message: error.message });
+  }
+};
+
+const getUserProducts = async (req, res) => {
```

Comment:
In my opinion, this code is too complicated. If we discharge the part that user can search according to title, we can use UserModel to find received, requested and donated - it will be much cleaner. 
But if we keep it like this, I think we can also add filtering according to category.

Reply:
I think you're right about the code not being clean and as you said if we get rid of the search it'll be cleaner, but issue